pkg/engine/datasource/graphql_datasource/graphql_datasource_test.go

```diff
@@ -2457,15 +2457,15 @@ func TestGraphQLDataSource(t *testing.T) {
 						Variables: resolve.NewVariables(
 							&resolve.ContextVariable{
 								Path:     []string{"title"},
-								Renderer: resolve.NewJSONVariableRendererWithValidation(`{"type":"string"}`),
+								Renderer: resolve.NewJSONVariableRendererWithValidation(`{"type":["string"]}`),
 							},
 							&resolve.ContextVariable{
 								Path:     []string{"completed"},
-								Renderer: resolve.NewJSONVariableRendererWithValidation(`{"type":"boolean"}`),
+								Renderer: resolve.NewJSONVariableRendererWithValidation(`{"type":["boolean"]}`),
 							},
 							&resolve.ContextVariable{
 								Path:     []string{"name"},
-								Renderer: resolve.NewJSONVariableRendererWithValidation(`{"type":"string"}`),
+								Renderer: resolve.NewJSONVariableRendererWithValidation(`{"type":["string","null"]}`),
```

Comment:
@jensneuse this is one of the places I was confused on what the code was supposed to be doing.  I made the test match the current functionality, but maybe it's the functionality that needs to be changed:

The issue here is that the AddTask mutation says that `name` is a non-nullable string.  But `$name` is _used_ in `input:[{user:{name: $name}}]`, and AddTaskInput.UserRef.name is a nullable string.  The code that constructs the renderer (`resolveNestedArgumentType`) is looking for the type of the input-object and getting the nullable string.

I suspect that this really should be a non-nullable string, and the logic in `resolveNestedArgumentType` is not actually right in this case.  But I don't really understand what's going on here.  I'm hoping you do, and know what the right fix is! (If indeed something needs to be fixed.)

Reply:
(There are a few other tests that have the same issue, but this one is as clear as any of them.)